openhtf/output/web_gui/__init__.py

```diff
@@ -357,21 +359,24 @@ def make_msg(cls, test_uid, remote_state):
 
   @classmethod
   def publish_test_state_update(cls, hostport, test_uid, state):
-    """Publish test update for to relevant subscribed clients."""
+    """Publish test update to relevant subscribed clients."""
     cls.publish(
         cls.make_msg(test_uid, state),
         client_filter=lambda c: cls.subscriber_to_hostport_map[c] == hostport)
 
   def on_subscribe(self, info):
-    """Add the subscriber and send intial state."""
+    """Add the subscriber and send initial state."""
     hostport = Hostport(info.arguments['host'][0],
                         int(info.arguments['port'][0]))
     self.subscriber_to_hostport_map[self] = hostport
+
     if hostport not in self._store.stations:
       _LOG.debug('Client tried to subscribe to unknown station. This can '
                  'happen as a result of the web gui server being restarted.')
       return
+
     try:
+      self._store.watch_tests(hostport)
       for test_uid, remote_test in self._store[hostport].tests.iteritems():
         self.send(self.make_msg(test_uid, remote_test.state))
```

Comment:
Technically this is still a race condition—the Angular frontend may receive the test state before the TestWatchers are started—but I don't think it will be an issue in practice